### PR TITLE
CRM-15849 add where clause for greetings

### PR DESCRIPTION
**Pull request by lcdservices**

### CODE REVIEW

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -3282,6 +3288,21 @@ public function sortName(&$values) {
     }
   }
 
+  /*
+   * Where/qill clause for greeting fields
+   *
+   * @params $values
```
Comment (colemanw) [inline CRM/Contact/BAO/Query.php]: Should be `@param array $values`

File: CRM/Contact/BAO/Query.php [hunk 2]
```diff
@@ -3282,6 +3288,21 @@ public function sortName(&$values) {
...
+   *
+   * @return void
+   */
+  function greetings(&$values) {
+    list($name, $op, $value, $grouping, $wildcard) = $values;
+    $name .= '_display';
+
+    $this->_qill[$grouping][] = "Greeting $op $value";
```
Comment (colemanw) [inline CRM/Contact/BAO/Query.php]: How to translate this string? Perhaps `ts('Greeting %1 %2', array(1 => $op, 2 => $value))` or maybe just `$name $op $value`?

**Comment by lcdservices** [comment]
I always forget the translation strings... sorry.
Let's keep it "Greeting %1 %2" -- the $name comes as lowercase values with underscores (addressee, email_greeting, postal_greeting), so we'd have to mess with it before inserting. I think the generic Greeting is sufficient.

**Comment by colemanw** [comment]
Thanks, looks good.

**Pull request merged**